File: TableGames.Games/card.js
define(['knockout', 'tmpl!card'], function(ko, cardTemplateName) {

    var suitNames = { 0: 'unknown', 1: 'clubs', 2: 'hearts', 3: 'spades', 4: 'diamonds' };
    var rankSymbols = { 0: '', 1: 'A', 2: '2', 3: '3', 4: '4', 5: '5', 6: '6', 7: '7', 8: '8', 9: '9', 10: '10', 11: 'J', 12: 'Q', 13: 'K' };

    function Card(cardState) {
        var self = this;

        self.isExposed = ko.observable(cardState.isExposed);
        self.suit = ko.observable(cardState.suit);
        self.rank = ko.observable(cardState.rank);
        self.isHighlighted = ko.observable(cardState.isHighlighted);

        self.suitName = ko.computed(function() { return suitNames[self.suit()]; });
        self.rankSymbol = ko.computed(function() { return rankSymbols[self.rank()]; });
    }
    Card.templateName = cardTemplateName;

    return Card;
});

File: TableGames.Games/Card.cs
namespace TableGames.Games
{
    public enum CardSuit
    {
        Unknown,
        Clubs,
        Hearts,
        Spades,
        Diamonds
    }

    public enum CardRank
    {
        Unknown,
        Ace,
        Two,
        Three,
        Four,
        Five,
        Six,
        Seven,
        Eight,
        Nine,
        Ten,
        Jack,
        Queen,
        King
    }

    public class Card
    {
        public CardSuit Suit { get; set; }
        public CardRank Rank { get; set; }
        public bool IsExposed { get; set; }
        public bool IsHighlighted { get; set; }
        public int Value {
            get {
                if (Rank == CardRank.Ace) {
                    return 14;
                }
                return (int)Rank;
            }
        }

        public int ValueA1 {
            get {
                return (int)Rank;
            }
        }

        public Card(CardSuit suit, CardRank rank) {
            Suit = suit;
            Rank = rank;
            IsExposed = false;
        }

        public object ToClient(bool includeValue = false) {
            return new {
                isExposed = IsExposed,
                suit = IsExposed || includeValue ? Suit : CardSuit.Unknown,
                rank = IsExposed || includeValue ? Rank : CardRank.Unknown,
                isHighlighted = IsHighlighted
            };
        }
    }
}